src/datasets/arrow_dataset.py

```diff
@@ -5620,7 +5620,8 @@ def push_to_hub(
         )
         repo_id = repo_url.repo_id
 
-        if revision is not None:
+        if revision is not None and not api.revision_exists(repo_id, revision, repo_type="dataset", token=token):
```

Comment:
Alternatively:
```suggestion
        if revision is not None and not revision.startswith("refs/pr/"):
```

Reply:
I think this modification would be more efficient:
- The modification above will fix the 400 Bad Request error without the additional network call `revision_exists`.
- In the case of a 403 Forbidden error, anyway we will raise some code lines below when trying to push to the branch...

Reply:
On the other hand, it is true that the call to `revision_exists` is a more general solution.